include/seastar/util/process.hh

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2022 Kefu Chai ( tchaikov@gmail.com )
+ */
+
+#pragma once
+
+#include <seastar/core/iostream.hh>
+
+namespace seastar {
+
+class process {
+public:
+    process(pid_t pid, int stdout, int stderr);
+    input_stream<char>& stdout() {
+        return _stdout;
+    }
+    input_stream<char>& stderr() {
+        return _stderr;
+    }
+    template<typename T, typename Tag>
+    struct tagged_type {
+        T value;
+    };
+    using exit_status = tagged_type<int, struct exit_status_tag>;
+    using term_sig = tagged_type<int, struct term_sig_tag>;
+    using wait_status = std::variant<exit_status, term_sig>;
+    future<wait_status> wait();
+private:
+    const pid_t _pid;
+    input_stream<char> _stdout;
+    input_stream<char> _stderr;
+};
+
+future<process> spawn_process(std::string_view pathhname,
+                              const std::vector<std::string_view>& argv,
```

Comment:
Why not removing the `pathname` completely and assuming that argv[0] is the binary name itself? Like python's subprocess.Popen does

Reply:
yes. this would simplify the API. actually, i was struggling when it came to the interface of `spawn_process()` in two perspectives:

1. shall we combine stdout and stderr? namely to `adddup2()` both `STDOUT_FILENO` and `STDERR_FILENO` to a single fd. this would simplify the error handling and interface a lot. but the downside is that there might be users want to separate stderr from stdout.
2. use different `pathname` and `argv[0]`? again, i was thinking of a "niche market", where user might need to use tools like "BusyBox". it is a single monolithic executable which expose different unix utilities to end user by checking `argv[0]`.

i might be overthinking the requirement. but at least, these are the reason why the interface looks this way.

Reply:
i think the new API (partially) addresses this usability issue.

Reply:
I think we shouldn't try to improve posix_spawn, just adapt it to Seastar.